crates/fluvio/src/producer/record.rs

```diff
@@ -0,0 +1,33 @@
+use dataplane::Offset;
+use fluvio_types::PartitionId;
+use async_channel::Receiver;
+
+use crate::error::Result;
+
+use super::error::ProducerError;
+
+#[derive(Clone, Debug, Default)]
+pub struct RecordMetadata {
```

Comment:
I think this is too heavy, it should be light as possible.  Probably some kind of a simple token or hash